test/spec/Radicle/Internal/ConcurrentMapTest.hs

```diff
@@ -0,0 +1,46 @@
+module Radicle.Internal.ConcurrentMapTest
+    ( test_
+    ) where
+
+import           Control.Concurrent.Async
+import           Protolude
+
+import qualified Radicle.Internal.ConcurrentMap as CMap
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+test_ :: TestTree
+test_ = testGroup "Radicle.Internal.ConcurrentMap"
```

Comment:
It would be nice to have a test for a `CMap Bool Int` with some `threadDelay`s which show (because of total time elapsed) that there is no contention if the key is `/=`, but I'm fine to merge as is.

Reply:
I have added a test for this. Could you have another look?